relation_engine_server/api_versions/api_v1.py

```diff
@@ -90,22 +91,36 @@ def run_query():
                                             batch_size=batch_size,
                                             full_count=full_count)
         return flask.jsonify(resp_body)
+
     if 'stored_query' in flask.request.args or 'view' in flask.request.args:
         # Run a query from a query name
         # Note: we are maintaining backwards compatibility here with the "view" arg.
         # "stored_query" is the more accurate name
         query_name = flask.request.args.get('stored_query') or flask.request.args.get('view')
         stored_query = spec_loader.get_stored_query(query_name)
-        stored_query_source = _preprocess_stored_query(stored_query['query'], stored_query)
+
+        has_ws_ids = False
+        if 'ws_ids' in stored_query['query']:
+            json_body['ws_ids'] = ws_ids
+            has_ws_ids = True
+        else:
+            del json_body['ws_ids']
```

Comment:
If the stored query doesn't have `ws_ids` in it, the additional `ws_ids` param will not be needed.

Reply:
As I mentioned in [this issue](https://github.com/kbase/relation_engine/issues/11), I think a simpler way to achieve the same thing with less code is to just remove line 75 here: https://github.com/kbase/relation_engine/blob/develop/relation_engine_server/api_versions/api_v1.py#L75

Reply:
If the query doesn't use `ws_ids`, it is just potentially slowing things down by retrieving the list of IDs and adding them to the query. The cleanest solution would be to check whether the query/stored query mentions `ws_ids`, and only if it does, fetch the IDs from the workspace to add to the request object and alter the AQL to include the ` LET ws_ids = @ws_ids ` line.

I've reverted things to how they were, with the `ws_ids` injection, but moved the setting of `ws_ids` to after JSON validation.

Reply:
Avoiding an extra workspace request when the query does not use ws_ids is a good point, I agree that would be a good additional benefit

Reply:
I'm seeing the current version of the code doesn't try to avoid the workspace request, and also doesn't remove line 75. Maybe you haven't pushed yet?

Reply:
OK, I have edited it to only add `ws_ids` and pull the list from the workspace when `ws_ids` is mentioned in the query.